### ISSUE TITLE
continue server on linux doesn't start or vscode extensions hangs after #575

### ISSUE THREAD

**Issue by lijunsong** [continuedev/continue#599]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [x] I've seen the [troubleshooting guide](https://continue.dev/docs/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: Ubuntu 20.04
- Continue: any versions after 0.1.30 (I think it's 0.1.30 because that's the first version command that has `continue.sendMainUserInput` command, which is merged in #575)
- IDE: vscode 1.80
```

### Description

After https://github.com/continuedev/continue/pull/575, the linux version doesn't work anymore in a sense that:
1. entering `/edit` (or actually anything) hangs the editor completely
2. the config.py doesn't load, so typing `/` in the text area doesn't have any response.

- Also it seems anything closer to 1.41 can't start the continue server (the initialization doesn't happen, continue.log is empty)
- Anything > =1.30 and <= 1.33 can start continue server but no command, entering commands hangs the editor
- anything before the #575 can start continue server and commands loads OK

The working version I know is 0.1.2x and before.

(I also tried 0.2.0 and above. The server doesn't start, so the GUI spins infinitely. I think another issue has reported the problem of 0.2.0 and above already.)

### To reproduce

1. On linux, install anything after 0.1.30 (I've tested all versions ranging from 0.1.16 - 0.1.41)
2. open the side panel, wait for loading
3. server starts OK, meilisearch starts OK
4. the config.py's commands are not preloaded; entering anything shows no type ahead and no response (or even freeze).

### Log output

```Shell
[2023-11-02 16:59:22,926] [DEBUG] Accepted websocket connection from Address(host='127.0.0.1', port=59006)
[2023-11-02 16:59:23,107] [DEBUG] Received message while initializing workspaceDirectory
[2023-11-02 16:59:23,108] [DEBUG] {'workspaceDirectory': '/home/user'}
[2023-11-02 16:59:23,108] [DEBUG] {'messageType': 'workspaceDirectory', 'data': {'workspaceDirectory': '/home/user'}}
[2023-11-02 16:59:23,108] [DEBUG] Received message while initializing uniqueId
[2023-11-02 16:59:23,108] [DEBUG] {'uniqueId': '47f5379a-1d78-49d1-88ab-0e7b5946391b'}
[2023-11-02 16:59:23,109] [DEBUG] {'messageType': 'uniqueId', 'data': {'uniqueId': '47f5379a-1d78-49d1-88ab-0e7b5946391b'}}
[2023-11-02 16:59:23,109] [DEBUG] []
[2023-11-02 16:59:23,176] [WARNING] Error sending IDE message, websocket probably closed: Unexpected ASGI message 'websocket.send', after sending 'websocket.close'.
[2023-11-02 16:59:23,923] [DEBUG] New session: None
[2023-11-02 16:59:23,927] [DEBUG] Loaded Continue config file from /home/user/.continue/config.py
[2023-11-02 16:59:23,951] [DEBUG] Starting context manager
[2023-11-02 16:59:23,989] [DEBUG] Sending session id: 0bdef43f-595b-441a-b81b-ad7d70666a3c
[2023-11-02 16:59:23,993] [DEBUG] Loading Meilisearch index...
[2023-11-02 16:59:24,016] [DEBUG] Loaded 0 documents into meilisearch in 0.002216815948486328 seconds for context provider code
[2023-11-02 16:59:24,019] [DEBUG] Loaded 1 documents into meilisearch in 0.005242586135864258 seconds for context provider diff
[2023-11-02 16:59:24,020] [DEBUG] Loaded 1 documents into meilisearch in 0.006244182586669922 seconds for context provider terminal
[2023-11-02 16:59:24,021] [DEBUG] Loaded 1 documents into meilisearch in 0.007572174072265625 seconds for context provider url
[2023-11-02 16:59:24,676] [WARNING] Error sending IDE message, websocket probably closed: Unexpected ASGI message 'websocket.send', after sending 'websocket.close'.
[2023-11-02 16:59:26,927] [WARNING] Error sending IDE message, websocket probably closed: Unexpected ASGI message 'websocket.send', after sending 'websocket.close'.
[2023-11-02 16:59:30,303] [WARNING] Error sending IDE message, websocket probably closed: Unexpected ASGI message 'websocket.send', after sending 'websocket.close'.
[2023-11-02 16:59:35,365] [WARNING] Error sending IDE message, websocket probably closed: Unexpected ASGI message 'websocket.send', after sending 'websocket.close'.
[2023-11-02 16:59:42,174] [DEBUG] Error loading meilisearch index: 
[2023-11-02 16:59:42,175] [DEBUG] Loaded Meilisearch index
[2023-11-02 16:59:44,015] [DEBUG] Error loading meilisearch index: 
[2023-11-02 16:59:44,017] [WARNING] Failed to kill process on port 7700 (access denied)
[2023-11-02 16:59:44,034] [DEBUG] Meilisearch started
[2023-11-02 16:59:44,078] [DEBUG] Loaded 0 documents into meilisearch in 0.00124359130859375 seconds for context provider code
[2023-11-02 16:59:44,082] [DEBUG] Loaded 1 documents into meilisearch in 0.005072116851806641 seconds for context provider diff
[2023-11-02 16:59:44,086] [DEBUG] Loaded 1 documents into meilisearch in 0.008512735366821289 seconds for context provider terminal
[2023-11-02 16:59:44,086] [DEBUG] Loaded 1 documents into meilisearch in 0.008627891540527344 seconds for context provider url
```
```

**Comment by sestinj** [comment]
@lijunsong There should be a binary for the server located at `~/.vscode/extensions/continue.continue-<version>/exe/continue_server`. If you double click this or run it from the terminal, does it successfully start or is there an error thrown? I'm wondering if this is a platform incompatibility problem introduced by any newer dependency

**Comment by lijunsong** [comment]
Oh correct, 0.1.39 has this error

```
[1030173] Error loading Python lib '/tmp/_MEIP7DjfC/libpython3.11.so.1.0': dlopen: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.35' not found (required by /tmp/_MEIP7DjfC/libpython3.11.so.1.0)
```

So the server requires higher GLIBC version, which means it was compiled on at least ubuntu 22.04. Ubuntu 20.04 has glibc2.31.

Sigh.

**Comment by sestinj** [comment]
"Sigh" is a very valid reaction

The Ubuntu version was changed in GitHub actions to support a new dependency that requires a newer version of sqlite, but I'm sure there's a way to manually package the newer sqlite even on 20.04

If you're looking to solve this faster than I might be able to fix the problem, you can manually run the server with ’pip install continuedev && python -m continuedev’.

**Comment by sestinj** [comment]
@lijunsong Excited to share that we no longer use the Python binary, which means that Continue is entirely compatible with any operating system!

I'll close this issue soon, but would love to know if things are back up and working for you : )
